app/controllers/requests_controller.rb

```diff
@@ -134,4 +135,79 @@ def request_params
     params.require(:request).permit(:name, :cpu_cores, :ram_mb, :storage_mb, :operating_system,
                                     :port, :application_name, :comment, :rejection_information, :status, user_ids: [], sudo_user_ids: [])
   end
+
+  # TODO get credentials and config for currently logged in user
+  # Clones and configures a git repository on dir.
+  def setup_git(dir)
+    # Dir.mkdir(dir) unless File.exists?(dir)
+    uri = "git@github.com:mewitte/PuppetFiles.git"
+    name = "PuppetFiles"
+
+    g = Git.clone(uri, name, :path => dir)
+    user = "Melvin Witte"
+    email = "melvin.witte@student.hpi.de"
+    g.config('user.name', user)
+    g.config('user.email', email)
+    g
+  end
+
+  # Creates node_vmname.pp. If file exists, overwrite file with potentially newer content
+  def write_node_script(dir, request)
+    puppet_string = puppet_script(request)
+    path = dir + '/Node/node_' + request.name + ".pp"
+    File.delete(path) if File.exists?(path)
+    File.open(path, 'w') { |f| f.write(puppet_string) } unless path.exists?
+  end
+
+  # TODO file not yet created
+  # Creates vmname.pp
+  def write_name_script
+  end
+
+  # TODO logic to change init.pp for new users
+  # Adapts init.pp for potential new users
+  def change_init_script
+  end
+
+  # Creates puppet files for request and pushes the created files into a git repository
+  def push_to_git(request)
+    dir = "temp_git_dir"
+    g = setup_git(dir)
+
+    write_node_script(dir, request)
+    write_name_script
+    change_init_script
+
+    if g.status.untracked.length + g.status.changed.length == 0
+      g.commit_all("Add node " + request.name)
+      g.push
+    end
+
+    FileUtils.rm_rf(dir)
```

Comment:
cloning and deleting might be a lot of work later for a single script if the repository becomes large

Reply:
Why is it necessary to delete the repo afterward? Would it add a lot of extra work to keep it around?
Performance is honestly not an issue for this application, because VMs are likely created at a very low rate.

Reply:
No, but you need to talk to the customer it's okay to keep the repository.